src/app/calendar-app/runbox-calendar-event.ts

```diff
@@ -69,137 +88,437 @@ export class RunboxCalendarEvent implements CalendarEvent {
+    // ?? Does this shorten it to one day ? (does it include the date)
+    // If set on an exception, only changes that exception
+    // FIXME: Does this change an ICAL.Time with no date (isDate=true) to
+    // one with a time? (as a side effect that is)
+    // See also set dtstart comments!
     set dtend(value: moment.Moment) {
         const allDay = this.allDay;
-        this.event.endDate = value ? ICAL.Time.fromJSDate(value.toDate()) : undefined;
+        if (value) {
+            this._dtend = this.momentToIcalTime(value, this.event.endDate ? this.event.endDate.zone : undefined);
+            // FIXME: (see above)
+            this.event.endDate = this._dtend;
+        }
         this.allDay = allDay;
     }
 
     // angular-calendar compatibility
 
     get start(): Date {
-        return this.dtstart.toDate();
+        return this._dtstart.toJSDate();
     }
 
     get end(): Date {
-        if (!this.dtend) {
+        if (!this._dtend) {
             return undefined;
         }
 
-        const shownEnd = moment(this.dtend);
+        const shownEnd = this._dtend.clone();
+        // ICAL event DTEND is exclusive, angular-calendar is inclusive
         if (this.allDay) {
-            shownEnd.subtract(1, 'days');
+            shownEnd.addDuration(new ICAL.Duration({'isNegative': true, 'days': 1}));
         } else {
-            shownEnd.subtract(1, 'seconds');
+            shownEnd.addDuration(new ICAL.Duration({'isNegative': true, 'seconds': 1}));
         }
-        return shownEnd.toDate();
+        return shownEnd.toJSDate();
     }
 
     get allDay(): boolean {
         // isDate in ICAL.Event means "has no time"
-        return this.event.startDate.isDate;
+        return this._dtstart.isDate;
+    }
+
+    get recurs(): boolean {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        return recur ? true : false;
     }
 
-    // deprecated, we should move away from this eventually
-    get rrule(): RRule {
-        let rrule = this.event.component.getFirstPropertyValue('rrule');
-        if (rrule) {
-            rrule = rrule.toString(); // ICAL claims this should be a string, but sometimes it's not
-            return rrulestr(rrule, { dtstart: this.dtstart.toDate() });
+    set recurs(flag: boolean) {
+        if (flag) {
+            const recur = this.event.component.getFirstPropertyValue('rrule');
+            if (!recur) {
+                this.event.component.addPropertyWithValue('rrule', new ICAL.Recur({ freq: 'DAILY' }));
+            }
+        } else {
+            this.event.component.removeProperty('rrule');
         }
-        return undefined;
     }
 
+    // This is the original DTSTART of the event - if recurring, will
+    // be different from dtstart above.
+    get recurStart(): Date {
+        return this.event.startDate.toJSDate();
+    }
+
+    set recurStart(start: Date) {
+        // Nope can't do that Dave
```

Comment:
Maybe log the attempt anyway? Unless this will never happen due to a user action.

Reply:
Nothing in this code attempts to set it - does ts have the concept of a read-only attribute ?

Reply:
I think that's what happens when you specify a `get` only.

Reply:
Huh I thought it didnt compile without - seems it does, so I'll remove it